### PR TITLE
Update shield_advanced.tf

### PR DESCRIPTION
**Pull request by eladlevi-dy**
Add the option to use "delete_unused_fm_managed_resources" on shield_advanced.

## what

- Terraform resource "aws_fms_policy" has the optional "delete_unused_fm_managed_resources" argument but the module does not.
- [delete_unused_fm_managed_resources](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/fms_policy#delete_unused_fm_managed_resources) - (Optional) If true, Firewall Manager will automatically remove protections from resources that leave the policy scope. Defaults to false.

## why

This change is needed because if resource is out of scope we don't want to keep using shield and gain extra unneeded costs

## references
[hashicorp - Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/fms_policy#delete_unused_fm_managed_resources)

### CODE REVIEW

File: shield_advanced.tf [hunk 1]
```diff
@@ -11,13 +11,14 @@ module "shield_advanced_label" {
...
-  name                        = module.shield_advanced_label[each.key].id
-  delete_all_policy_resources = lookup(each.value, "delete_all_policy_resources", true)
-  exclude_resource_tags       = lookup(each.value, "exclude_resource_tags", false)
-  remediation_enabled         = lookup(each.value, "remediation_enabled", false)
-  resource_type_list          = lookup(each.value, "resource_type_list", null)
-  resource_type               = lookup(each.value, "resource_type", null)
-  resource_tags               = lookup(each.value, "resource_tags", null)
+  name                               = module.shield_advanced_label[each.key].id
+  delete_all_policy_resources        = lookup(each.value, "delete_all_policy_resources", true)
+  delete_unused_fm_managed_resources = lookup(each.value, "delete_unused_fm_managed_resources", false)
```
Comment (milldr) [inline shield_advanced.tf:16]: please add this value to the description of the variable
https://github.com/cloudposse/terraform-aws-firewall-manager/blob/main/variables.tf#L137

Comment (eladlevi-dy) [reply shield_advanced.tf:16]: @milldr - Done.

**Pull request merged**